tsql/TSqlParser.g4

```diff
@@ -3204,6 +3204,7 @@ function_call
     | aggregate_windowed_function                       #AGGREGATE_WINDOWED_FUNC
     | analytic_windowed_function                        #ANALYTIC_WINDOWED_FUNC
     | scalar_function_name '(' expression_list? ')'     #SCALAR_FUNCTION
+    | STRING_AGG '(' aggcol=expression ',' (expression | STRING) ')' (WITHIN GROUP '(' order_by_clause ')')?  #STRINGAGG
```

Comment:
Useless alternative: `expression | STRING`. Because of `expression` already includes `STRING` by such way: `expression -> primitive_expression -> constant -> STRING.

Reply:
Oops, let me remove and retest.